lib/exercises.rb

```diff
@@ -2,21 +2,47 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n)
+# Space Complexity: o(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  # raise NotImplementedError, "Method hasn't been implemented yet!"
+  hash = {}
+  strings.each do |string|
+    sorted_string = string.split("").sort.join
+    if hash[sorted_string]
+      hash[sorted_string] << string
+    else 
+      hash[sorted_string] = [string]
+    end
+  end
+  return hash.values
 end
 
+
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n)
+# Space Complexity: o(n)
 def top_k_frequent_elements(list, k)
```

Comment:
This doesn't actually work (see below) and is O(k * n log n) in time complexity as you're sorting the hash k times.